code/game/sound.dm

```diff
@@ -288,118 +314,134 @@ var/list/rummage_sound = list(\
 
 	return toplay
 
-/proc/playsound(var/atom/source, soundin, vol as num, vary, extrarange as num, falloff, var/is_global, var/use_pressure = TRUE)
```

Comment:
The replacement of use_pressure with frequency here may cause compatibility issues with things that call playsound(); it'd be worth going through and checking every single use of playsound if you're wanting to replace the sound code with baycode's sound code wholesale.
For reference, `zrange`, `override_env`, `envdry`, and `envwet` are all that's necessary for zsound (the multiz sound implementation in the PR linked in your OP) to work.

Reply:
We've been having a lot of sound issues over the years, honestly it might be worth replacing wholesale. I'll do some thorough testing to see what gets broken and what doesn't.